### ISSUE TITLE
Reconcile will not stop if the watched dependent resource has PersistentVolumeClaim

### ISSUE THREAD

**Issue by wzhanw** [operator-framework/operator-sdk#4906]
## Bug Report
Reconcile never stop if the watched dependent resource has a PersistentVolumeClaim

{"level":"debug","ts":1620702396.9731293,"logger":"predicate","msg":"Reconciling due to dependent resource update","name":"wml-conda","namespace":"wmla","apiVersion":"v1","kind":"PersistentVolumeClaim"}
{"level":"debug","ts":1620702396.991792,"logger":"predicate","msg":"Reconciling due to dependent resource update","name":"wmla-cond","namespace":"wmla","apiVersion":"v1","kind":"PersistentVolumeClaim"}

#### What did you do?
I used an ansible operator to manage k8s deployment, secret, services and PVC.

#### What did you expect to see?
After the PersistentVolumeClaim created by the ansible operator, the operator should ignore the update if it bound the PV,

#### What did you see instead? Under which circumstances?

Reconcile loop will not stop if there is PVC in the watched depenedant resources and PVC bind a PV.

#### Environment

**Operator type:**
/language ansible

**Kubernetes cluster type:**
OpenShift 4.6.16

`$ operator-sdk version`
1.7.2

`$ kubectl version`

Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0+e49167a", GitCommit:"e49167aad6a08046be6ab21ff13029110c76951d", GitTreeState:"clean", BuildDate:"2021-01-28T07:35:27Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

**Comment by estroz** [comment]
Is your operator watching the PV or PVC, or are you saying the dependent resource uses a PVC and PVCs are not explicitly watched? If the dependent resource changes because a PVC change, reconcile will be triggered; this is how controllers work.

/triage support

**Comment by kensipe** [comment]
@wzhanw thanks filing the issue!   Is it possible for you to provide more detail on the PVC configuration and any additional logs?  It would be incredibly helpful to have more log detail or a way to reproduce the situation... thanks!

**Comment by blakebarnett** [comment]
I believe I'm hitting this also, everything was working fine until I added PVCs to the deployments that the operator is handling. Is it necessary to add resource types to the `NewControllerManagedBy()` call in addition to deployment here? `Owns(&appsv1.Deployment{}).`

I'm confused why it worked fine with secretVolumeSources and configMapVolumeSources but not this. Let me know if there any specifics that I can provide to help debug (other than the code which I can't post unfortunately).

**Comment by blakebarnett** [comment]
I did a few tests around this and have determined that adding the PVC was a red-herring. This operator manages a set of between 3 and 5 deployments cluster-wide in namespaces for each set. 

- Everything works fine when I reduce it to 2 deployments (each with an accompanying configmap, secret, service, ingress).
-  If I add a 3rd, Reconcile loops infinitely. 
- If I keep the 3rd one that was added and remove the 2nd, it works fine again. So it's nothing unique to any of the deployments and it's other resources. 

Is this triggering some limitation of the framework once the # of resources the Reconcile loop is handling gets past a certain threshold? (The number of deployments/namespaces being managed didn't seem to be related)
